### ISSUE TITLE
execution error: Finder got an error: Can’t get disk "Test DMG". (-1728)

### ISSUE THREAD

**Issue by sainianil** [andreyvit/create-dmg#43]
Sometimes, I'm getting this error and background image was not set for dmg when this error occurs.

**execution error: Finder got an error: Can’t get disk "Test DMG". (-1728)**

**DMG script log:**

```
./create-dmg --volname 'Test DMG' --volicon Icon.icns --background DMGBackground.png --window-pos 200 120 --window-size 800 600 --icon-size 100 --icon 'Test DMG.app' 220 410 --app-drop-link 580 410 'Test DMG.dmg' 
/Users/workspace/Test DMG.xcarchive/Products/Applications/
Creating disk image...
created: /Users/workspace/Utilities/InstallerScript/rw.Test DMG.dmg
Mounting disk image...
Unmounting disk image...
Mount directory: /Volumes/Test DMG
Device name:     /dev/disk1
Copying background file...
making link to Applications dir
/Volumes/Test DMG
Copying volume icon file 'Icon.icns'...
Running Applescript: /usr/bin/osascript **"/var/folders/sr/vch_50x104g461x64sbys3hc0000gn/T/createdmg.fI8bJ9Bm" "Test DMG"
/var/folders/sr/vch_50x104g461x64sbys3hc0000gn/T/createdmg.fI8bJ9Bm:85:89: execution error: Finder got an error: Can’t get disk "Test DMG". (-1728)**
Done running the applescript...
Fixing permissions...
Done fixing permissions.
Blessing started
Blessing finished
Unmounting disk image...
"disk1" unmounted.
"disk1" ejected.
Compressing disk image...
Preparing imaging engine…
Reading Protective Master Boot Record (MBR : 0)…
   (CRC32 $F4037986: Protective Master Boot Record (MBR : 0))
Reading GPT Header (Primary GPT Header : 1)…
   (CRC32 $BE188168: GPT Header (Primary GPT Header : 1))
Reading GPT Partition Data (Primary GPT Table : 2)…
   (CRC32 $055CBC10: GPT Partition Data (Primary GPT Table : 2))
Reading  (Apple_Free : 3)…
   (CRC32 $00000000:  (Apple_Free : 3))
Reading disk image (Apple_HFS : 4)…
   (CRC32 $330AA36B: disk image (Apple_HFS : 4))
Reading  (Apple_Free : 5)…
   (CRC32 $00000000:  (Apple_Free : 5))
Reading GPT Partition Data (Backup GPT Table : 6)…
   (CRC32 $055CBC10: GPT Partition Data (Backup GPT Table : 6))
Reading GPT Header (Backup GPT Header : 7)…
   (CRC32 $235739C8: GPT Header (Backup GPT Header : 7))
Adding resources…
Elapsed Time:  3.121s
File size: 4566212 bytes, Checksum: CRC32 $6A7A7124
Sectors processed: 57344, 20356 compressed
Speed: 3.2Mbytes/sec
Savings: 84.4%
created: /Users/workspace/Utilities/InstallerScript/Test DMG.dmg
hdiutil: internet-enable: enable succeeded
Disk image done
```

**Comment by kaloyan-raev** [comment]
This happens quite often to me too. My workaround is to configure Jenkins to recognize the "-1728" error in the log and run the job once again.

**Comment by sainianil** [comment]
Thanks Kaloyan-raev, I'm also doing the same workaround as of now however I want to know the cause of this error so that we can fix this error.

**Comment by seltzered** [comment]
Been building with create-dmg for years and never experienced this error till today, running High Sierra 10.13.2 on apfs. Workaround was to do the same thing of just running it again.

**Comment by aonez** [comment]
This must be fixed with #55. Reopen if not.
